content/en/docs/guides/flux-e2e.md

```diff
@@ -0,0 +1,330 @@
+
+### 7. `GitRepository` Source (Artifacts and Revisions)
+
+A `GitRepository` is a Custom Resource that saves a read-only view of the latest revision of a Git repository and hosts it as a service in the cluster.
+
+The [GitRepository Custom Resource][] describes a (usually remote) Git repository, including the URL of the repository host, and the Git reference (such as a branch name or tag) to monitor for changes. Additionally you may find a secret reference with SSH or TLS keys to verify that host, a secret reference containing authentication credentials, a secret reference containing keys for verifying commit signatures, a configurable polling interval, and other meta information related to the source repository.
+
+The Source Controller treats its connection to each Git repository as read-only, even if the authentication credentials supplied would enable writes to the repository. (Other components, such as the Image Update Automation Controller, may use the Git repository for its access and can write to the repository.)
+
+The Source Controller connects to the repository host, pulls the latest commit for the specified reference, and stores the contents in a bundled and compressed format (currently a gzipped tarball file).
+
+The Source Controller also supports:
+* the [BucketSpec Custom Resource][] for reading the contents of an Amazon S3, Google Cloud Storage and other similar block-storage services;
+* the [HelmRepository Custom Resource][] for reading the index of a Helm chart repository, and
+* the [HelmChart Custom Resource][] for reading a Helm chart artifact from a HelmRepository source.
+
+Note that it does not make any difference to the Source Controller whether a source is hosted within the cluster or on an external service or server. The Source Controller will still attempt to verify the host using the SSH or TLS keys supplied in the GitRepository Custom Resource and will store its contents as a read-only tarball.
+
+Features include:
+* Validate source definitions
+* Authenticate to sources (SSH, user/password, API token)
+* Validate source authenticity (PGP)
+* Detect source changes based on update policies (semver)
+* Fetch resources both on-demand (via webhooks) and on-a-schedule (at a configured polling interval)
+* Package the fetched resources into a well-known format (tar.gz, yaml)
+* Make the artifacts addressable by their source identifier (sha, version, ts)
+* Make the artifacts available in-cluster to interested 3rd parties (such as the Kustomize Controller and Helm Controller)
+* Notify interested 3rd parties of source changes and availability (status conditions, events, hooks)
```

Comment:
Quite liked this, makes it easy to refer to the component's main features/responsibilities. :+1: 
Maybe worth having a shorter version here and a link to the up-to-date information (SC's readme?) that may change over time.

Reply:
We can take things out in a subsequent PR, and move it elsewhere 👍